### PR TITLE
cf-runagent exit code based on remote agent runs results

### PR DESCRIPTION
**Pull request by vpodzime**

### CODE REVIEW

**Comment by vpodzime** [comment]
> jenkins: [Build Status](https://ci.cfengine.com/job/pr-pipeline/6305/)

selenium tests retry: [Build Status](https://ci.cfengine.com/job/testing-run-php-selenium-tests/1567/)
sequential tests retry: [Build Status](https://ci.cfengine.com/job/sequential-tests/1734/)

**Review by larsewi — DISMISSED** [review]
Nice !

File: cf-runagent/cf-runagent.c [hunk 1]
```diff
@@ -158,6 +160,41 @@ char REMOTEBUNDLES[CF_MAXVARSIZE] = "";
...
+        else if (WIFEXITED(remote_exit_status))
+        {
+            *exit_code = WEXITSTATUS(remote_exit_status);
+            return;
+        }
+        else
+        {
+            *exit_code = CF_RA_EXIT_CODE_OTHER_ERR;
+            return;
+        }
```
Comment (olehermanse) [inline cf-runagent/cf-runagent.c:187]: Nitpick, but it's easier for me to see that this part is guaranteed to return if you skip the elses:

```suggestion
        if (WIFEXITED(remote_exit_status))
        {
            *exit_code = WEXITSTATUS(remote_exit_status);
            return;
        }
        *exit_code = CF_RA_EXIT_CODE_OTHER_ERR;
        return;
```

**Review by olehermanse — APPROVED** [review]
Great improvement! 👍

File: cf-serverd/server_common.c [hunk 2]
```diff
@@ -1790,7 +1792,15 @@ bool DoExec2(const EvalContext *ctx,
         }
     }
     free(line);
-    cf_pclose(pp);
+    int exit_code = cf_pclose(pp);
+    if (exit_code >= 0)
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply cf-serverd/server_common.c:1796]: Some error in `cf_pclose()` (which should never happen).

**Review by olehermanse — APPROVED** [review]

**Pull request merged**